test/hotspot/gtest/runtime/test_virtualMemoryTracker.cpp

```diff
@@ -69,22 +104,18 @@ static void check_inner(ReservedMemoryRegion* rmr, R* regions, size_t regions_si
 class VirtualMemoryTrackerTest {
 public:
   static void test_add_committed_region_adjacent() {
```

Comment:
Reviewer hint: the point of these and similar changes throughout `VirtualMemoryTrackerTest` is to prevent the test from modifying global NMT state. IMHO gtests should not do this since it leads to all sorts of problems. In this case, I rewrite the tests to not modify, but just react on NMT state. And then, in test/hotspot/jtreg/gtest/NMTGtests.java, I start the NMT subset of these tests with NMT enabled.

This gives me better coverage and prevents weird test errors from NMT state changing during VM life. It also makes it possible to run the whole gtest suite with NMT enabled and test NMT as well as use NMT to monitor memory usage of the gtests.